common/flatpak-dir.c

```diff
@@ -1486,6 +1538,50 @@ flatpak_dir_system_helper_call_ensure_repo (FlatpakDir   *self,
   return ret != NULL;
 }
 
+static gboolean
+flatpak_dir_system_helper_call_get_revokefs_fd (FlatpakDir   *self,
+                                                guint         arg_flags,
+                                                const gchar  *arg_ref,
+                                                const gchar  *arg_commit,
+                                                const gchar  *arg_installation,
+                                                gint         *out_socket,
+                                                gchar       **out_src_dir,
+                                                GCancellable *cancellable,
+                                                GError      **error)
+{
+  g_autoptr(GUnixFDList) out_fd_list = NULL;
+  gint fd = -1;
+  gint fd_index = -1;
+
+  if (flatpak_dir_get_no_interaction (self))
+    arg_flags |= FLATPAK_HELPER_GET_REVOKEFS_FD_FLAGS_NO_INTERACTION;
+
+  g_debug ("Calling system helper: GetRevokefsFd");
+
+  g_autoptr(GVariant) ret =
+    flatpak_dir_system_helper_call (self, "GetRevokefsFd",
+                                    g_variant_new ("(usss)",
+                                                   arg_flags,
+                                                   arg_ref,
+                                                   arg_commit,
```

Comment:
How does this work if arg_commit is NULL? This function is called with opt_commit, i.e. it can be NULL, and this will both break when doing this call, but also is not handled properly in the system helper, where it assumes its always set.

Reply:
With re-use any cache-repo if available, we really don't need this now. Earlier we need this commit arg to find the `.commitparital` thing.

Same goes for `ref` as we don't need to have per-ref cache-repo. Both are stripped down from DBus arguments in the recent iteration.